### PR TITLE
Add experiment.info.significance event.

### PR DESCRIPTION
**Pull request by romain-growthbook**
### Features and Changes

- Closes **(add link to issue here)**

### CODE REVIEW

File: packages/back-end/src/services/experimentNotifications.ts [hunk 1]
```diff
@@ -170,16 +192,206 @@ export const notifySrm = async ({
...
+export const notifySignificance = async ({
+  context,
+  experiment,
+  snapshot: currentSnapshot,
+}: {
+  context: Context;
+  experiment: ExperimentInterface;
+  snapshot: ExperimentSnapshotDocument;
+}) => {
+  const lastSnapshot = await getLatestSnapshot(
```
Comment (romain-growthbook) [inline packages/back-end/src/services/experimentNotifications.ts:270]: Makse sure that this is executed _before_ saving a new snaopshot!

File: packages/back-end/src/services/experimentNotifications.ts [hunk 2]
```diff
@@ -170,16 +192,206 @@ export const notifySrm = async ({
...
+    experiment.id,
+    experiment.phases.length - 1
+  );
+
+  if (!lastSnapshot) return;
```
Comment (romain-growthbook) [inline packages/back-end/src/services/experimentNotifications.ts:275]: Cases:
1. Something broken
2. First snapshot for the experiment.

Test case: make sure we fire on the first snapshot if it is significant on the first run.

File: packages/back-end/src/services/experimentNotifications.ts (hunk 1 above)
Comment (lukesonnet) [reply packages/back-end/src/services/experimentNotifications.ts]: (instead we now use dateCreated of the snapshots)

File: packages/back-end/src/events/handlers/slack/slack-event-handler-utils.ts [hunk 3]
```diff
@@ -371,6 +377,56 @@ const buildSlackMessageForExperimentDeletedEvent = (
...
+    metricName,
+    variationName,
+    experimentName,
+  }: {
+    metricName: string;
+    variationName: string;
+    experimentName: string;
+  }) => {
+    if (statsEngine === "frequentist") {
+      return `In experiment ${experimentName}: metric ${metricName} for variation ${variationName} is
```
Comment (lukesonnet) [inline packages/back-end/src/events/handlers/slack/slack-event-handler-utils.ts:401]: Make experimentName here and in the "bayesian" case below a link to the experiment with the #results anchor.

File: packages/back-end/src/services/experimentNotifications.ts [hunk 4]
```diff
@@ -95,11 +103,18 @@ export const notifyAutoUpdate = ({
...
-        success,
-        experimentId: experiment.id,
-        experimentName: experiment.name,
+      dispatchEvent({
+        context,
+        experiment,
+        event: "warning",
+        data: {
+          object: {
+            type: "auto-update",
```
Comment (lukesonnet) [inline packages/back-end/src/services/experimentNotifications.ts:112]: auto-update isn't really a "warning" event type. Is this necessary to classify it as such?

**Review by lukesonnet — APPROVED** [review]
tested in local for email and slack, and unit tests look correct logic wise.

furthermore took a closer look at the more important parts (e.g. `getLatestSnapshot` changes) and it lgtm

File: packages/back-end/src/events/handlers/slack/slack-event-handler-utils.ts [hunk 5]
```diff
@@ -288,6 +294,11 @@ const buildSlackMessageForFeatureDeletedEvent = async (
 export const getExperimentUrlFormatted = (experimentId: string): string =>
   `\n• <${APP_ORIGIN}/experiment/${experimentId}|View Experiment>`;
 
+export const getExperimentUrlAndNameFormatted = (
+  experimentId: string,
+  experimentName: string
+): string => `<${APP_ORIGIN}/experiment/${experimentId}|${experimentName}>`;
```
Comment (romain-growthbook) [inline packages/back-end/src/events/handlers/slack/slack-event-handler-utils.ts:300]: `APP_ORIGIN` was missing indeed! Also didn't realized this was not a `markdown` formatted string..

**Pull request merged**